todo_app/templates/table_macro.html

```diff
@@ -19,34 +19,32 @@
               {# Below sets the variable pos (for workflow position) which is used in identifying which status icon to load #}
               {# If first status then 0 (ToDo), if last status then 2 (Done) for any intermediate status then 1 (In Progress) #}
               
-              {% set statusesId_list = statuses.keys()|list%}
-              {% if statusesId_list[0] == item.status %} {%set pos=0%}
-              {% elif statusesId_list.index(item.status) == statusesId_list|length-1 %} {%set pos=2%}
+              
+              {% if item.status == statuses.0 %} {%set pos=0%}
+              {% elif item.status == statuses|last %} {%set pos=2%}
               {% else %} {%set pos=1%}
               {% endif %}
```

Comment:
If this is so unintuitive that you felt a code comment was required, why not write it differently.
E.g.

```
              {% if item.status == statuses.0 %} {%set status_icon='/static/status_0.png'%}
              {% elif item.status == statuses|last %} {%set status_icon='/static/status_2.png'%}
              {% else %} {%set set status_icon='/static/status_1.png'%}
              {% endif %}
```

You could actually make the png names more helpful too.

Reply:
Very good suggestion. Changed and also made changes to the png name